add adapter x-touch

Bannsaenger opened the pull request:
Adapter added with the suggested method (npm run addToLatest ...).
devices sort order changed by the script.

Apollon77 commented:
RE-CHECK!

GermanBluefox commented:
## Automated adapter checker

### [ioBroker.vivitek](https://github.com/Bannsaenger/ioBroker.vivitek)
[![Downloads](https://img.shields.io/npm/dm/iobroker.vivitek.svg)](https://www.npmjs.com/package/iobroker.vivitek) ![Number of Installations (latest)](http://iobroker.live/badges/vivitek-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.vivitek.png?downloads=true)](https://nodei.co/npm/iobroker.vivitek/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "vivitek" in latest repository

### [ioBroker.devices](https://github.com/ioBroker/ioBroker.devices)
[![Downloads](https://img.shields.io/npm/dm/iobroker.devices.svg)](https://www.npmjs.com/package/iobroker.devices) ![Number of Installations (latest)](http://iobroker.live/badges/devices-installed.svg) ![Number of Installations (stable)](http://iobroker.live/badges/devices-stable.svg)
[![NPM](https://nodei.co/npm/iobroker.devices.png?downloads=true)](https://nodei.co/npm/iobroker.devices/)

- [ ] :heavy_exclamation_mark: [E150] No common.connectionType found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)
- [ ] :heavy_exclamation_mark: [E152] No common.dataSource found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)

### [ioBroker.x-touch](https://github.com/Bannsaenger/ioBroker.x-touch)
[![Downloads](https://img.shields.io/npm/dm/iobroker.x-touch.svg)](https://www.npmjs.com/package/iobroker.x-touch) ![Number of Installations (latest)](http://iobroker.live/badges/x-touch-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.x-touch.png?downloads=true)](https://nodei.co/npm/iobroker.x-touch/)

- [ ] :heavy_exclamation_mark: [E605] No actual year found in copyright. Please add "Copyright (c) 2021 Bannsaenger <bannsaenger@gmx.de>" at the end of [README.md](https://github.com/Bannsaenger/ioBroker.x-touch/blob/master/README.md)
- [ ] :heavy_exclamation_mark: [E701] No actual year found in LICENSE. Please add "Copyright (c) 2021 Bannsaenger <bannsaenger@gmx.de>" at the start of LICENSE
- [ ] :eyes: [W400] Cannot find "x-touch" in latest repository


*Add comment "RE-CHECK!" to start check anew*

Bannsaenger commented:
RE-CHECK!

GermanBluefox commented:
## Automated adapter checker

### [ioBroker.devices](https://github.com/ioBroker/ioBroker.devices)
[![Downloads](https://img.shields.io/npm/dm/iobroker.devices.svg)](https://www.npmjs.com/package/iobroker.devices) ![Number of Installations (latest)](http://iobroker.live/badges/devices-installed.svg) ![Number of Installations (stable)](http://iobroker.live/badges/devices-stable.svg)
[![NPM](https://nodei.co/npm/iobroker.devices.png?downloads=true)](https://nodei.co/npm/iobroker.devices/)

- [ ] :heavy_exclamation_mark: [E150] No common.connectionType found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)
- [ ] :heavy_exclamation_mark: [E152] No common.dataSource found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)

### [ioBroker.x-touch](https://github.com/Bannsaenger/ioBroker.x-touch)
[![Downloads](https://img.shields.io/npm/dm/iobroker.x-touch.svg)](https://www.npmjs.com/package/iobroker.x-touch) ![Number of Installations (latest)](http://iobroker.live/badges/x-touch-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.x-touch.png?downloads=true)](https://nodei.co/npm/iobroker.x-touch/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "x-touch" in latest repository

### [ioBroker.vivitek](https://github.com/Bannsaenger/ioBroker.vivitek)
[![Downloads](https://img.shields.io/npm/dm/iobroker.vivitek.svg)](https://www.npmjs.com/package/iobroker.vivitek) ![Number of Installations (latest)](http://iobroker.live/badges/vivitek-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.vivitek.png?downloads=true)](https://nodei.co/npm/iobroker.vivitek/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "vivitek" in latest repository


*Add comment "RE-CHECK!" to start check anew*

Bannsaenger commented:
Devices wurde von dem addToLatest umsortiert. Kann ich an dieser Stelle leider nicht ändern.

Bannsaenger commented:
RE-CHECK!

GermanBluefox commented:
## Automated adapter checker

### [ioBroker.vivitek](https://github.com/Bannsaenger/ioBroker.vivitek)
[![Downloads](https://img.shields.io/npm/dm/iobroker.vivitek.svg)](https://www.npmjs.com/package/iobroker.vivitek) ![Number of Installations (latest)](http://iobroker.live/badges/vivitek-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.vivitek.png?downloads=true)](https://nodei.co/npm/iobroker.vivitek/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "vivitek" in latest repository

### [ioBroker.devices](https://github.com/ioBroker/ioBroker.devices)
[![Downloads](https://img.shields.io/npm/dm/iobroker.devices.svg)](https://www.npmjs.com/package/iobroker.devices) ![Number of Installations (latest)](http://iobroker.live/badges/devices-installed.svg) ![Number of Installations (stable)](http://iobroker.live/badges/devices-stable.svg)
[![NPM](https://nodei.co/npm/iobroker.devices.png?downloads=true)](https://nodei.co/npm/iobroker.devices/)

- [ ] :heavy_exclamation_mark: [E150] No common.connectionType found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)
- [ ] :heavy_exclamation_mark: [E152] No common.dataSource found in [io-package.json](https://github.com/ioBroker/ioBroker.devices/blob/master/io-package.json)

### [ioBroker.x-touch](https://github.com/Bannsaenger/ioBroker.x-touch)
[![Downloads](https://img.shields.io/npm/dm/iobroker.x-touch.svg)](https://www.npmjs.com/package/iobroker.x-touch) ![Number of Installations (latest)](http://iobroker.live/badges/x-touch-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.x-touch.png?downloads=true)](https://nodei.co/npm/iobroker.x-touch/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "x-touch" in latest repository

### [ioBroker.device-reminder](https://github.com/xenon-s/ioBroker.device-reminder)
[![Downloads](https://img.shields.io/npm/dm/iobroker.device-reminder.svg)](https://www.npmjs.com/package/iobroker.device-reminder) ![Number of Installations (latest)](http://iobroker.live/badges/device-reminder-installed.svg) ![Number of Installations (stable)](http://iobroker.live/badges/device-reminder-stable.svg)
[![NPM](https://nodei.co/npm/iobroker.device-reminder.png?downloads=true)](https://nodei.co/npm/iobroker.device-reminder/)

- [ ] :heavy_exclamation_mark: [E503] "[iob_npm.done](https://github.com/xenon-s/ioBroker.device-reminder/blob/master/iob_npm.done)" found in repo! Please add it to .gitignore
- [ ] :heavy_exclamation_mark: [E605] No actual year found in copyright. Please add "Copyright (c) 2021 xenon-s <ente_s@hotmail.de>" at the end of [README.md](https://github.com/xenon-s/ioBroker.device-reminder/blob/master/README.md)
- [ ] :heavy_exclamation_mark: [E701] No actual year found in LICENSE. Please add "Copyright (c) 2021 xenon-s <ente_s@hotmail.de>" at the start of LICENSE
- [ ] :heavy_exclamation_mark: [E818] file iob_npm.done found in repository, but not found in .npmignore
- [ ] :heavy_exclamation_mark: [E914] file iob_npm.done found in repository, but not found in .gitignore


*Add comment "RE-CHECK!" to start check anew*

Bannsaenger commented:
Sorry for that. 
Ich kann die device Adapter leider nicht ändern.

Apollon77 commented:
alles gut, ignorieren

Apollon77 commented:
Hi,

sorry for the late review, here the comments:
* you use adapter-core 2.4.0, so io-package should contain a js-controller dep to >=2.0.0
* vivitek: when you destroy/disconnect the client in unload ... will the reconnection logic be applied? Please test that in compact mode!
* x-touch: basically same question then above ... also: close should have a callbacl which is cslled when closed or? Better use this to trigger unload callback
* extron: is end allowing a callback? ;-) (unload again)

Thank you for a check here

Apollon77 commented:
@Bannsaenger Is there anything where we can support?

Bannsaenger commented:
Sorry for the late reply,
io-package fixed.
vivitek: changed destroy to end. end accepts a callback
x-touch: added callback to server.close
extron: ssh2.client does not have the ability to take a callback.
The reconnect behaviour of extron will be subject to change in a newer version.
Hope that the adapters could be checked in.

Apollon77 commented:
Great and welcome to the adapter repository!
